### PR TITLE
build: add build targets for NATS and add NATS configuration

### PR DESCRIPTION
**Pull request by lenny-intel**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-service-configurable/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A*
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [x] I have opened a PR for the related docs change (if not, why?) 
  https://github.com/edgexfoundry/edgex-docs/pull/846

## Testing Instructions
run `make build-nats`
Verify `-tags "include_nats_messaging"` is in the `go build` command
run  `make docker-nats`
Verify `-tags "include_nats_messaging"` is in the `go build` Step for the docker image

### CODE REVIEW

**Review by AlexCuse — DISMISSED** [review]

**Review by AlexCuse — DISMISSED** [review]

**Review by jim-wang-intel — DISMISSED** [review]

File: makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.PHONY: build test clean docker
+.PHONY: build build-nats test clean docker docker-nats vendor
```
Comment (ernestojeda) [inline makefile:1]: What are these new PHONY targets used for? `build-nats` `docker-nats`

Comment (lenny-intel) [reply makefile:1]: For got to pull them out.

**Review by ernestojeda — APPROVED** [review]
One small comment about the update to `.PHONY`

File: makefile (hunk 1 above)
Comment (lenny-intel) [reply makefile]: Removed those and add tidy

**Review by jim-wang-intel — APPROVED** [review]

**Review by jim-wang-intel — APPROVED** [review]

**Pull request merged**